pkg/roachpb/metadata.go

```diff
@@ -199,7 +199,11 @@ func (r RangeDescriptor) String() string {
 	} else {
 		buf.WriteString("<no replicas>")
 	}
-	fmt.Fprintf(&buf, ", next=%d, gen=%d]", r.NextReplicaID, r.Generation)
+	gen := int64(-1)
+	if r.Generation != nil {
+		gen = *r.Generation
+	}
```

Comment:
I think you just want to use `r.GetGeneration()` here. A nil `Generation` is considered 0 (not -1) and is how all new range descriptors start.